chttpd.c

```diff
@@ -57,6 +58,61 @@ request_handler(void *cls, struct MHD_Connection *connection, const char *url,
 }
 
 
+void
+parse_port(const char *port_arg, int *port) {
+    bool is_valid = true;
+
+    for (int i = 0; i < strlen(port_arg); i++) {
+        if (port_arg[i] < '0' || port_arg[i] > '9') {
+            is_valid = false;
+            break;
+        }
+    }
+
+    if (is_valid == false) {
+        ERROR("Invalid port number: %s\n", port_arg);
+        exit(EXIT_FAILURE);
+    }
+
+    *port = atoi(port_arg);
+}
+
+
+void
+parse_bind_addr(const char *bind_addr_arg, char *bind_addr, int *port) {
+    int colon_pos = -1;
+
+    if (strcmp(bind_addr_arg, "localhost") == 0) {
+        bind_addr_arg = "127.0.0.1";
```

Comment:
@pylover please ensure it is good enough to only convert `localhost` to its relevant IP address.